lib/ktl/cluster.rb

```diff
@@ -104,5 +114,13 @@ def reassignment_progress
     def execute_reassignment(reassigner, plan, dryrun = false)
       ReassignmentTask.new(reassigner, plan, shell, logger: logger).execute(dryrun)
     end
+
+    def create_reassigner(zk_client, options)
+      if options.wait?
+        ContinousReassigner.new(zk_client, limit: options.limit, logger: logger, log_assignments: options.verbose, delay: options.delay, shell: shell)
+      else
+        Reassigner.new(zk_client, limit: options.limit, logger: logger, log_assignments: options.verbose)
+      end
```

Comment:
```ruby
implementation = options.wait? ContinousReassigner : Reassigner
implementation.new(zk_client, limit: options.limit, logger: logger, log_assignments: options.verbose)
```

or even move this to a method on `Reassigner` and have it return different instances depending on the value of an option.